### ISSUE TITLE
how to calculate the macs after prune?

### ISSUE THREAD

**Issue by machengjie321** [pytorch/pytorch#100859]
### 🚀 The feature, motivation and pitch

I use torch.nn.utils.prune as prune to prune the model, then I use torchprofile.profile_macs() to calculate the macs of Pruned_model, but I find the macs will be increase before prune.remove() to make the pruning permanent. it is normal because additional calculate wil be weight * mask.
but after I use prune.remove() to make the pruning permanent, the macs calculated by torchprofile.profile_macs() still same as the model befor prune.I use torch.nn.utils.prune as prune to prune the model, then I use torchprofile.profile_macs() to calculate the macs of Pruned_model, but I find the macs will be increase before prune.remove() to make the pruning permanent. it is normal because additional calculate wil be weight * mask.
but after I use prune.remove() to make the pruning permanent, the macs calculated by torchprofile.profile_macs() still same as the model befor prune.
![2023-05-08_17-01-34](https://user-images.githubusercontent.com/38120691/236770558-5905bf8d-6c72-4de6-afef-26c373273e37.jpg)

cc @jerryzh168 @jianyuh @raghuramank100 @jamesr66a @vkuzo @jgong5 @Xia-Weiwen @leslie-fang-intel

**Comment by jcaip** [comment]
Hi @machengjie321

I don't believe `torch.nn.utils.prune` offers any performance benefits, it just zeros out the weights. 

Can you print out the weight shape after you call `prune.remove()`? If it's the same as before, then this should be expected behavior.

**Comment by machengjie321** [comment]
> Hi @machengjie321
> 
> I don't believe `torch.nn.utils.prune` offers any performance benefits, it just zeros out the weights.
> 
> Can you print out the weight shape after you call `prune.remove()`? If it's the same as before, then this should be expected behavior.

Hi@jcaip
Thanks for your Reply.
Yes, torch.nn.utils.prune does not any performance benefits. After i use prune.ln_structured(), since weight*mask will generate more flops, the flops after prune will increase. After I remove the mask, the mask will be removed and the pruned weight will transforme to zero, so the additional weight*mask will  disappear, the flops will be same as when it is not prune.
At this time, the weights shape is the same as before, excepet that the pruned weight become zero.
![1687833813399](https://github.com/pytorch/pytorch/assets/38120691/6457984e-5dc1-48ac-bebc-9b91ecb35bd9)
![1687833859838](https://github.com/pytorch/pytorch/assets/38120691/2f2bdf9c-b06e-496e-8b58-91fd8172d886)

**Comment by jcaip** [comment]
@machengjie321 I see, thanks for checking for me. Yeah in this case unfortunately there is no way to get less MACs with the `nn.utils.prune` flow. 

You can try to use our experimental structured pruning flow [here](https://github.com/pytorch/pytorch/blob/main/torch/ao/pruning/_experimental/pruner/README.md). 

If your model is FX traceable you can use this flow to get a pruned model that will be faster (but only for row-wise pruning).
